native-sql-engine/cpp/src/codegen/arrow_compute/ext/window_kernel.cc

```diff
@@ -544,6 +544,316 @@ arrow::Status WindowRankKernel::AreTheSameValue(const std::vector<ArrayList>& va
   return arrow::Status::OK();
 }
 
+WindowLagKernel::WindowLagKernel(
```

Comment:
TODO: the cpp file is already too big, might split this into small files

Reply:
Will refactor it in another patch. Thanks!